### PR TITLE
copy the test_perf_attrib testcase of empyrical 0.5.2/ fix some code to support python3 

### PR DESCRIPTION
**Pull request by huaiweicheng**

### CODE REVIEW

**Comment by twiecki** [comment]
Thanks @huaiweicheng . I'm actually working on similar fixes here: https://github.com/quantopian/pyfolio/pull/608 Maybe wait until that is merged and then see what is left over.

**Comment by twiecki** [comment]
@huaiweicheng OK, I think that branch is in a stable state. I changed the test environment to use more recent python and pandas so the tests break due to that (i.e. the Panel issue) https://travis-ci.org/quantopian/pyfolio/jobs/575223398. If you still want to help with that it'd be much appreciated. I think the best approach would be to work off of my branch, and then do a PR onto that branch. We know when it's done when the tests for 3.7 there pass. Then, we can then merge your PR into my mine and then mine into master. Does that make sense?

**Comment by huaiweicheng** [comment]
@twiecki  That's fine. So be it.

**Comment by twiecki** [comment]
FYI I changed the base to merge into fix_deprs instead. Seems like there are merge conflicts. It might be easier to start fresh from that branch.

**Comment by huaiweicheng** [comment]
Hi @twiecki  You can review the code and check whether the refactor way is fine. Since I have no style exposure of AAPL & ABT of the US market. I fake the momentum&size style data in the comment.

**Comment by twiecki** [comment]
@huaiweicheng This looks good to me, however, note the merge conflicts preventing us from merging this. What I would advise is to either start a new branch off of my `fix_deprs` and manually reapply the code changes from here, or try to cherry-pick your commits on top of that branch.

**Comment by huaiweicheng** [comment]
Ok. I will push merging the conflicts in the first place and continue to do new submits.

File: pyfolio/plotting.py [hunk 1]
```diff
@@ -645,7 +645,7 @@ def show_perf_stats(returns, factor_returns=None, positions=None,
         perf_stats = pd.DataFrame(perf_stats_all, columns=['Backtest'])
 
     for column in perf_stats.columns:
-        for stat, value in perf_stats[column].iteritems():
+        for stat, value in perf_stats[column].items():
```
Comment (twiecki) [inline pyfolio/plotting.py]: I think this needs to stay as it's a Series, not a dict.

File: pyfolio/tears.py [hunk 2]
```diff
@@ -1347,7 +1358,7 @@ def create_perf_attrib_tear_sheet(returns,
             ax=plt.subplot(gs[current_section])
         )
 
-    gs.tight_layout(fig)
+    # gs.tight_layout(fig)
```
Comment (twiecki) [inline pyfolio/tears.py]: Why?

**Comment by twiecki** [comment]
@huaiweicheng Let me know when you think this is done, it largely looks OK to me but see small comments. Not sure why we don't get travis building here.

File: pyfolio/plotting.py (hunk 1 above)
Comment (huaiweicheng) [reply pyfolio/plotting.py]: cool

File: pyfolio/tears.py (hunk 2 above)
Comment (huaiweicheng) [reply pyfolio/tears.py]: The tight_layout will make the legend majority part out of scope while savefig into pdf or png when I ran the tear function.

**Comment by huaiweicheng** [comment]
@twiecki  ok. I will compelete all the changes in about one week and when it's done I will let you konw.

**Comment by huaiweicheng** [comment]
@twiecki I believe I have done almostlty. You have any suggestions?

**Comment by twiecki** [comment]
@huaiweicheng I'll merge this into this other PR to hope that it triggers the automatic build.

**Comment by huaiweicheng** [comment]
It's werid that the travis does not be triggered  :(

**Comment by twiecki** [comment]
@huaiweicheng Well we have something to work with here: https://travis-ci.org/quantopian/pyfolio/builds/577715297?utm_source=github_status&utm_medium=notification starting with pep8 fixes. Want to do another PR after branching off of the current `fix_deprs`?

**Comment by huaiweicheng** [comment]
@twiecki  Ok. I'll do it.

**Comment by huaiweicheng** [comment]
@twiecki I have open a new PR #613.  You can megre it into fix_deprs

**Pull request merged**